llvm/lib/Target/SPIRV/SPIRVGlobalRegistry.cpp

```diff
@@ -260,8 +260,18 @@ Register SPIRVGlobalRegistry::buildConstantFP(APFloat Val,
     MF.getRegInfo().setRegClass(Res, &SPIRV::IDRegClass);
     assignTypeToVReg(LLVMFPTy, Res, MIRBuilder);
     DT.add(ConstFP, &MF, Res);
-    MIRBuilder.buildFConstant(Res, *ConstFP);
+    if (EmitIR) {
+      MIRBuilder.buildFConstant(Res, *ConstFP);
+    } else {
```

Comment:
I'm not sure I follow the full context here:
Once this MIR is generated, all the instructions are lowers to actual SPIR-V instructions in the final printers.
So here, if we generate the FConstant or the OpConstantF, what difference does it makes?
Is it to make sure the SPIR-V types remains exactly the one we passed, and don't leave the choice to the code lowering MIR to final SPIR-V?

Reply:
Right.
So before it was called in only one location: `SPIRVBuiltins.cpp`, but this was before the final lowering, so calling `MIRBuilder.buildFConstant` was fine.
But now that you call it in the instruction selector which is later you need to generate this ?
Wondering if the only usage today could be replaced by the new code?
What if you directly generate the constant SPIR-V instruction in SPIRVBuilting?
Is that breaking something like some constant propagation pass?

Reply:
> But now that you call it in the instruction selector which is later you need to generate this ?

Yes

> Wondering if the only usage today could be replaced by the new code?
> What if you directly generate the constant SPIR-V instruction in SPIRVBuilting?
> Is that breaking something like some constant propagation pass?

That shouldn't be a problem. MIR and gMIR can be mixed and actually it looks like the one place `SPIRVBuiltin` uses the const is within a SPIR-V instruction so it might as well be a SPIR-V const.